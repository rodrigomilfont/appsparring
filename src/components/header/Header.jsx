import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import HeaderSearch from '../../containers/header/HeaderSearch';
import './index.css';

const Header = props => {
  const { showSearch } = props;
  return (
    <header className="header">
      <div className="box-logo">
        <Link to="/"> Logo </Link>
      </div>
      {showSearch && <HeaderSearch {...props} />}
    </header>
  );
};

Header.propTypes = {
  showSearch: PropTypes.bool,
};

Header.defaultProps = {
  showSearch: false,
};

export default Header;
