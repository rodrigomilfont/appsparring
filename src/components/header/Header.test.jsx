import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import sinon from 'sinon';
import Header from './Header';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', ()=> {
  const wrapper = shallow(<Header/>);
  // console.log(wrapper.debug())
  expect(wrapper).toMatchSnapshot();
});

it('change search term', ()=> {
  const onSearchTermChange = sinon.spy();
  const wrapper = shallow(<Header onSearchTermChange={onSearchTermChange} />);

  wrapper.find('input').simulate('change', { target: { value: "changed by test" } })
  
  expect(onSearchTermChange.calledOnce).toEqual(true);
});

it('submit to history', ()=> {
  const wrapper = shallow(<Header />)
  wrapper.setState({ searchTerm: "searchTermByTest" })
  wrapper.find('form').simulate('submit', { preventDefault () {} })
  //TODO Test history
});

