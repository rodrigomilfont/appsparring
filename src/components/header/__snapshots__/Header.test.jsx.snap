// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    showSearch={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="box-logo"
        >
          <Link
            replace={false}
            to="/"
          >
             Logo 
          </Link>
        </div>,
        false,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
          >
             Logo 
          </Link>,
          "className": "box-logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " Logo ",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": " Logo ",
          "type": [Function],
        },
        "type": "div",
      },
      false,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="box-logo"
          >
            <Link
              replace={false}
              to="/"
            >
               Logo 
            </Link>
          </div>,
          false,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
               Logo 
            </Link>,
            "className": "box-logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Logo ",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": " Logo ",
            "type": [Function],
          },
          "type": "div",
        },
        false,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
